src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -24,63 +27,87 @@ public EndpointParameter(Endpoint endpoint, IParameterSymbol parameter, WellKnow
         Source = EndpointParameterSource.Unknown;
         IsOptional = parameter.IsOptional();
         DefaultValue = parameter.GetDefaultValueString();
-        IsArray = TryGetArrayElementType(parameter, out var elementType);
+        IsArray = TryGetArrayElementType(Type, out var elementType);
         ElementType = elementType;
         IsEndpointMetadataProvider = ImplementsIEndpointMetadataProvider(parameter, wellKnownTypes);
         IsEndpointParameterMetadataProvider = ImplementsIEndpointParameterMetadataProvider(parameter, wellKnownTypes);
+        ProcessEndpointParameterSource(endpoint, parameter, parameter.GetAttributes(), wellKnownTypes);
+    }
 
-        if (parameter.HasAttributeImplementingInterface(wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromRouteMetadata), out var fromRouteAttribute))
+    public EndpointParameter(Endpoint endpoint, IPropertySymbol property, IParameterSymbol? parameter, WellKnownTypes wellKnownTypes)
```

Comment:
Can we de-dupe the ctors? Or somehow only have 1 ctor?

Reply:
I like the type safety that comes with two distinct constructors and the fact that it's clearer to delineate behaviors that impact one scenario but not another. For example, for non-AsParameters scenarios, we don't need to worry about coalescing attributes on the property and matched parameter. There's also the benefit of being able to make the constructor that takes a `IPropertySymbol` private since we can only initialize `AsParameters` attributes that way.

That being said, I think it's possible to create a shared base constructor here that captures the common elements.